Data/ByteString.hs

```diff
@@ -1735,11 +1735,9 @@ sort (BS input l)
 useAsCString :: ByteString -> (CString -> IO a) -> IO a
 useAsCString (BS fp l) action =
   allocaBytes (l+1) $ \buf ->
-    -- Cannot use unsafeWithForeignPtr, because action can diverge
-    withForeignPtr fp $ \p -> do
-      memcpy buf p l
-      pokeByteOff buf l (0::Word8)
-      action (castPtr buf)
+    unsafeWithForeignPtr fp $ \p -> memcpy buf p l
+    pokeByteOff buf l (0::Word8)
+    action (castPtr buf)
```

Comment:
Is there something wrong with the indentation or the linebreak?!

Reply:
I forgot `do`